### PR TITLE
Feature/update doctor profile

### PR DESCRIPTION
**Pull request by SwiftXLight**
**Description changes:** 
- add endpoint for update doctor info
- add guard for update endpoint

[link to Jira ticket 55](https://smartdc23.atlassian.net/browse/SMAR-55?atlOrigin=eyJpIjoiOWE1ZTM3ODliNTdkNDllYzgzZGNmZTk4ODUyMDRjYTIiLCJwIjoiaiJ9)
[link to Jira ticket 52](https://smartdc23.atlassian.net/browse/SMAR-52?atlOrigin=eyJpIjoiZmQxMmMyMGZlY2Q0NDVkMDgxZmU4MDdjNzVjNDU1MDUiLCJwIjoiaiJ9)

P.S.: branch created not from develop, but from feat/register+email branch

**PR checklist:**
1.1 Common
1.1.1 types for input and output params
1.1.2 no any, should be also added to eslint rules
1.1.3 try/catch for any async function
1.1.4 no magic numbers
1.1.5 compare only with constants not with strings
(instead of user === ‘admin’, better user === roles.admin)
1.1.6 no ternary operator inside ternary operator (bad example: question ? first:
second ? third: fourth)
1.1.7 avoid similar types with duplicating by generics
1.1.8 no console.log in pr
1.1.9 .env file should be in .gitignore
1.1.10 delete commented code if it's not part of documentation
1.1.11 no links in the code, env links should be in env file (for example: server url),
constant links (for example default avatar URL) should be in constant file
1.1.12 constants and function names should be lowercase.
1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant
variable
1.1.14 import rules. imports should come in a specific order: node modules, absolute
path, relative path
1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
const EACH_DAY_15h_15min = ‘15 3 * * *’

1.2 Backend
1.2.1 swagger for each endpoint
1.2.2 less requests to db, all data should be taken with one query
1.2.3 `public` in method use only is use function externally
1.2.4 use ConfigService instead of `process.env`

**Successful patch:**
image
image

**Patch failed because of wrong token:**
image

**No token error (user not logged in):**
image

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -49,14 +101,174 @@ export default class AuthService {
...
+
+    const doctor = await this.validateDoctorFromGoogle(googleDoctor);
+
+    const accessToken = this.jwtService.sign(
+      { ...doctor },
+      { expiresIn: '8h' },
+    );
+
+    res.cookie('accessToken', accessToken, this.accessTokenCookieOptions);
+    res.redirect('http://localhost:4200/');
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: its env

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -49,14 +101,174 @@ export default class AuthService {
...
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private async getGoogleUser({
+    id_token,
+    access_token,
+  }): Promise<GoogleDoctorResult> {
+    try {
+      const res = await axios.get(
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: env or constant

Comment (SwiftXLight) [reply src/modules/auth/auth.service.ts]: solved

File: src/modules/auth/auth.service.ts (hunk 1 above)
Comment (SwiftXLight) [reply src/modules/auth/auth.service.ts]: solved

File: src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
 import SMTPTransport from 'nodemailer/lib/smtp-transport';
+import UpdateGoogleDoctorDto from 'modules/doctor/dto/update-google-doctor-dto';
+import { Request, Response, CookieOptions } from 'express';
+import LoginDoctorDto from '../doctor/dto/login-doctor.dto';
+
+import CreateDoctorDto from '../doctor/dto/create-doctor.dto';
 import DoctorService from '../doctor/doctor.service';
 import Doctor from '../doctor/entity/doctor.entity';
-import LoginDoctorDto from '../doctor/dto/login-doctor.dto';
+import { GOOGLE_URL, HASH_NUMBER, SEVEN } from '../../shared/consts';
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: absolute path

File: src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -49,9 +109,146 @@ export default class AuthService {
...
+    );
+
+    res.cookie('accessToken', accessToken, this.accessTokenCookieOptions);
+    res.redirect(this.configService.get('CLIENT_URL'));
+  }
+
+  private async getGoogleOauthTokens(code: {
+    code: string;
+  }): Promise<Record<string, string>> {
+    const url = 'https://oauth2.googleapis.com/token';
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: I have seen this code today few times, it should be in env or constant

File: src/modules/doctor/doctor.service.ts [hunk 5]
```diff
@@ -51,6 +51,29 @@ export default class DoctorService {
...
+      throw new HttpException(
+        'User with this email already exists',
+        HttpStatus.CONFLICT,
+      );
+    }
+    return user;
+  }
+
+  async getDoctorByEmail(email: string): Promise<Doctor> {
+    const user = await this.doctorRepository
```
Comment (lyudad) [inline src/modules/doctor/doctor.service.ts]: why sometimes you have try/catch, sometimes no

File: src/modules/doctor/dto/update-google-doctor-dto.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  IsISO31661Alpha2,
+  IsNumber,
+} from 'class-validator';
+import {
+  CITY_REGEX,
+  ADDRESS_REGEX,
+  DATE_REGEX,
+  TIME_ZONE_REGEX,
+} from '../../../shared/consts';
+import { Role, Gender } from '../../../shared/enums';
```
Comment (lyudad) [inline src/modules/doctor/dto/update-google-doctor-dto.ts]: absolute path

Comment (SwiftXLight) [reply src/modules/doctor/dto/update-google-doctor-dto.ts]: fixed

File: src/modules/doctor/doctor.service.ts (hunk 5 above)
Comment (SwiftXLight) [reply src/modules/doctor/doctor.service.ts]: fixed

File: src/modules/auth/auth.service.ts (hunk 4 above)
Comment (SwiftXLight) [reply src/modules/auth/auth.service.ts]: add const

File: src/modules/auth/auth.service.ts (hunk 3 above)
Comment (SwiftXLight) [reply src/modules/auth/auth.service.ts]: fixed

File: src/modules/doctor/dto/update-google-doctor-dto.ts (hunk 6 above)
Comment (SwiftXLight) [reply src/modules/doctor/dto/update-google-doctor-dto.ts]: fixed

File: src/modules/auth/auth.service.ts (hunk 4 above)
Comment (SwiftXLight) [reply src/modules/auth/auth.service.ts]: add const

**Pull request merged**